internal/v4/auth.go

```diff
@@ -45,29 +45,19 @@ func (h *Handler) authorize(req *http.Request, acl []string) error {
 		}
 	}
 
-	// If basic auth credentials are presented, use them.
-	user, passwd, err := parseCredentials(req)
-	if err == nil {
-		if user != h.config.AuthUsername || passwd != h.config.AuthPassword {
-			return errgo.WithCausef(nil, params.ErrUnauthorized, "invalid user name or password")
-		}
-		return nil
-	}
-	if err != errNoCreds || h.store.Bakery == nil || h.config.IdentityLocation == "" {
-		return errgo.WithCausef(err, params.ErrUnauthorized, "authentication failed")
-	}
-
-	// No basic auth presented: use macaroon third-party caveat verification.
-	attrs, verr := httpbakery.CheckRequest(h.store.Bakery, req, nil, checkers.New())
+	admin, attrs, verr := h.checkRequest(req)
 	if verr == nil {
-		logger.Infof("authenticated with attrs %q", attrs)
+		logger.Infof("authenticated with admin: %q, attrs %q", admin, attrs)
+		if admin {
```

Comment:
This is another place where having Admin in the struct returned by checkRequest
would make things simpler (the check could be done in checkACLMembership)